client-app/src/admin/tests/fetch/FetchApiTestModel.js

```diff
@@ -0,0 +1,413 @@
+import { HoistModel, XH} from '@xh/hoist/core';
+import { bindable } from '@xh/hoist/mobx';
+
+
+@HoistModel
+export class FetchApiTestModel {
+
+    @bindable testServer;
+    @bindable testMethod;
+    @bindable response = null;
+    referenceSite = 'https://httpstatuses.com/';
+    testServers = [
+        {
+            value: '//localhost:8080/fetchTest?status=',
+            label: 'Grails - Localhost'
+        },
+        {
+            value: 'https://httpstat.us/',
+            label: 'httpstat.us'
+        }
+    ];
+    testMethods = [
+        {
+            value: 'fetch',
+            label: 'fetch (GET)'
+        }, 'getJson', 'postJson', 'putJson', 'deleteJson'
+    ]
+
+    constructor() {
+        this.setTestServer(this.testServers[0].value);
+        this.setTestMethod(this.testMethods[0].value);
+    }
+
+    async requestCodeAsync(code) {
+        switch (this.testMethod) {
+            case 'fetch': return this.doFetchASync(code);
+            default: return  XH.fetchService[this.testMethod]({
+                fetchOpts: {
+                    credentials: this.useCreds ? 'include' : 'omit'
+                },
+                url: `${this.testServer}${code}`,
+                headers: {
+                    'Expect': code == 100 ? '100-continue' : undefined
+                }
+            }).then(
+                (resp) => {
+                    console.log(resp);
+                    
+                    this.setResponse(JSON.stringify(resp, undefined, 2));
+                }, 
+                (err) => this.handleError(err)
+            );
+        }
+    }
+
+    async doFetchASync(code) {
```

Comment:
typo here in method name (ASync).  Curious about use of switch here, and the use of 2 arguments to then(), rather than catch